GitCommands/Git/GitModule.cs

```diff
@@ -2725,10 +2725,20 @@ public bool IsDirtyDir()
                 : null;
         }
 
+        /// <summary>
+        /// Get the file contents for the HEAD commit
+        /// </summary>
+        /// <param name="file">The Git status item</param>
+        /// <returns>An awaitable task with the requested file contents.</returns>
         public async Task<string?> GetFileContentsAsync(GitItemStatus file)
         {
-            var contents = new StringBuilder();
+            if (!file.IsTracked || (file.Staged == StagedStatus.Index && file.IsNew))
+            {
+                // File is not in Git
+                return null;
+            }
 
+            var contents = new StringBuilder();
```

Comment:
```suggestion
            StringBuilder contents = new();
```

Reply:
We could change this globally too
(this was just a moved line, I do normally not updated unrelated changes to keep diffs reviewable.)
Updated, no other changes than this and the squash